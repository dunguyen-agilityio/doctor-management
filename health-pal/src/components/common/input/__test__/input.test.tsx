import { fireEvent, render, screen } from '@utils-test'

import { X } from '@icons'

import { Input } from '..'

const MockIcon = () => <X testID="mock-icon" />

describe('Input', () => {
  const mockProps = {
    value: 'test',
    onChangeText: jest.fn(),
    placeholder: 'Enter text',
  }
  const mockErrorMessage = 'Invalid input'
  const mockOnFocus = jest.fn()
  const mockOnEndEdit = jest.fn()
  const mockOnEndEditing = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders without icon', () => {
    render(<Input {...mockProps} />)

    expect(screen.getByPlaceholderText('Enter text')).toBeTruthy()
    expect(screen.queryByTestId('left-icon')).toBeNull()
    expect(screen.queryByText(mockErrorMessage)).toBeNull()
    expect(screen.getByPlaceholderText('Enter text')).toHaveStyle({ paddingLeft: 16 })
  })

  it('matches snapshot without icon', () => {
    const { toJSON } = render(<Input {...mockProps} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders with icon', () => {
    render(<Input {...mockProps} leftIcon={MockIcon} />)

    expect(screen.getByTestId('mock-icon')).toBeTruthy()
  })

  it('matches snapshot with icon', () => {
    const { toJSON } = render(<Input {...mockProps} leftIcon={MockIcon} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders error message with red border', () => {
    render(<Input {...mockProps} errorMessage={mockErrorMessage} />)

    expect(screen.getByText(mockErrorMessage)).toHaveStyle({ color: 'red' })
  })

  it('calls onFocus and tracks value', () => {
    render(<Input {...mockProps} onFocus={mockOnFocus} />)

    fireEvent(screen.getByPlaceholderText('Enter text'), 'focus')
    expect(mockOnFocus).toHaveBeenCalled()
  })

  it('calls onEndEdit with isChanged based on value change', () => {
    render(<Input {...mockProps} onEndEdit={mockOnEndEdit} />)

    fireEvent(screen.getByPlaceholderText('Enter text'), 'focus')
    fireEvent(screen.getByPlaceholderText('Enter text'), 'endEditing', {
      nativeEvent: { text: 'new value' },
    })
    expect(mockOnEndEdit).toHaveBeenCalledWith(true)

    fireEvent(screen.getByPlaceholderText('Enter text'), 'focus')
    fireEvent(screen.getByPlaceholderText('Enter text'), 'endEditing', {
      nativeEvent: { text: 'test' },
    })
    expect(mockOnEndEdit).toHaveBeenCalledWith(false)
  })

  it('forwards onEndEditing prop', () => {
    render(<Input {...mockProps} onEndEditing={mockOnEndEditing} />)

    fireEvent(screen.getByPlaceholderText('Enter text'), 'endEditing', {
      nativeEvent: { text: 'test' },
    })
    expect(mockOnEndEditing).toHaveBeenCalled()
  })
})
