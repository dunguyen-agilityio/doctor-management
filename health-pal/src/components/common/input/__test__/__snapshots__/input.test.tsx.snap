// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot with icon 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={[]}
  >
    <View
      style={[]}
    >
      <View
        style={[]}
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height="24"
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          testID="mock-icon"
          vbHeight={24}
          vbWidth={24}
          width="24"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
          >
            <RNSVGPath
              d="M18 6 6 18"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            />
            <RNSVGPath
              d="m6 6 12 12"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <TextInput
        defaultValue="test"
        focusVisibleStyle={{}}
        focusable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        placeholder="Enter text"
        placeholderTextColor="#9ca3af"
        style={
          {
            "backgroundColor": "#f9fafb",
            "borderBottomColor": "#d1d5db",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderBottomWidth": 1,
            "borderLeftColor": "#d1d5db",
            "borderLeftWidth": 1,
            "borderRightColor": "#d1d5db",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "#d1d5db",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderTopWidth": 1,
            "color": "#374151",
            "flex": 1,
            "fontFamily": "Inter_400Regular",
            "fontSize": 14,
            "height": 45,
            "minWidth": 0,
            "paddingBottom": 12,
            "paddingLeft": 54,
            "paddingRight": 16,
            "paddingTop": 12,
            "position": "relative",
          }
        }
      />
    </View>
  </View>,
]
`;

exports[`Input matches snapshot without icon 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={[]}
  >
    <View
      style={[]}
    >
      <TextInput
        defaultValue="test"
        focusVisibleStyle={{}}
        focusable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        placeholder="Enter text"
        placeholderTextColor="#9ca3af"
        style={
          {
            "backgroundColor": "#f9fafb",
            "borderBottomColor": "#d1d5db",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderBottomWidth": 1,
            "borderLeftColor": "#d1d5db",
            "borderLeftWidth": 1,
            "borderRightColor": "#d1d5db",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "#d1d5db",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderTopWidth": 1,
            "color": "#374151",
            "flex": 1,
            "fontFamily": "Inter_400Regular",
            "fontSize": 14,
            "height": 45,
            "minWidth": 0,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
      />
    </View>
  </View>,
]
`;
